### ISSUE TITLE
web3.js: Remove use of Buffer

### ISSUE THREAD

**Issue by macalinao** [solana-labs/solana#27590]
#### Problem
Buffer isn't built for the web. Uint8Array is and saves 10kb gzipped.

#### Proposed Solution

Begin migrating code to use Uint8Array instead. Code that doesn't use Borsh should get optimized automatically.

**Comment by steveluscher** [comment]
100%. This is another change that we should negotiate with near/borsh-js to eliminate this dependency from the web bundle entirely, everywhere. cc/ @ailisp.

**Comment by ailisp** [comment]
Your treeshaking proposal is impressive! We also have plan to drop Buffer-usage in borsh. Uint8Array is available for all, <10-year browsers. The only challenge part is to borsh serialize a string, it uses `Buffer.from(string, 'utf-8')`. Standard web api for doing that is `TextEncoder`, but that need polyfills.

**Comment by macalinao** [comment]
> Your treeshaking proposal is impressive! We also have plan to drop Buffer-usage in borsh. Uint8Array is available for all, <10-year browsers. The only challenge part is to borsh serialize a string, it uses `Buffer.from(string, 'utf-8')`. Standard web api for doing that is `TextEncoder`, but that need polyfills.

Yeah Uint8Array has good polyfills so I think it's fine-- virtually all frontend packaging tools have some form of browserslist/polyfilling in their build configs.

10kb gzipped if you guys accomplish this. And no more browserify! 🙂
